ehitchhiking-ui/src/app/pages/main-screen/main-screen-component/main-screen.component.ts

```diff
@@ -58,9 +64,23 @@ export class MainScreenComponent implements OnInit {
 
 	toggleStateToPassenger() {
 	  this.userState = UserState.passenger;
+	  this.isPensilStateOfPlusButton = false;
```

Comment:
this.isPensilStateOfPlusButton = false;
    this.isShownViewRoutesButton = false;
    this.isNeedCleanMap = true;
    this.mapTriggers = {reset:true};

That code is common for toggleStateToPassenger and toggleStateToDriver hence you should move it to separate function

Reply:
done